src/SonarScanner.MSBuild.PreProcessor/WebService/SonarWebService.cs

```diff
@@ -349,43 +244,64 @@ private async Task<T> ExecuteWithLogs<T>(Func<Task<T>> request, Uri logUri)
             return settings;
         }
 
-        private void GetPropertyValue(Dictionary<string, string> settings, JToken p)
+        /// <summary>
+        /// Concatenates project key and branch into one string.
+        /// </summary>
+        /// <param name="projectKey">Unique project key</param>
+        /// <param name="projectBranch">Specified branch of the project. Null if no branch to be specified.</param>
+        /// <returns>A correctly formatted branch-specific identifier (if appropriate) for a given project.</returns>
+        protected Uri GetUri(string query, params string[] args) =>
+            new(serverUri, Escape(query, args));
+
+        protected virtual Uri AddOrganization(Uri uri, string organization) =>
+            string.IsNullOrEmpty(organization)
+                ? uri
+                : new Uri(uri + $"&organization={WebUtility.UrlEncode(organization)}");
+
+        private Dictionary<string, string> ParseSettingsResponse(string contents)
         {
-            var key = p["key"].ToString();
-            if (p["value"] != null)
+            var settings = new Dictionary<string, string>();
+            var settingsArray = JObject.Parse(contents).Value<JArray>("settings");
+            foreach (var t in settingsArray)
             {
-                var value = p["value"].ToString();
-                settings.Add(key, value);
+                GetPropertyValue(settings, t);
             }
-            else if (p["fieldValues"] != null)
+
+            return CheckTestProjectPattern(settings);
+        }
+
+        private IList<SensorCacheEntry> ParseCacheEntries(Task<Stream> task)
+        {
+            if (task.IsFaulted || task.Result is not { } dataStream)
             {
-                MultivalueToProps(settings, key, (JArray)p["fieldValues"]);
+                return new List<SensorCacheEntry>();
             }
-            else if (p["values"] != null)
+
+            var cacheEntries = new List<SensorCacheEntry>();
+            try
             {
-                var array = (JArray)p["values"];
-                var value = string.Join(",", array.Values<string>());
-                settings.Add(key, value);
+                while (dataStream.Position < dataStream.Length)
+                {
+                    cacheEntries.Add(SensorCacheEntry.Parser.ParseDelimitedFrom(dataStream));
+                }
             }
-            else
+            catch (Exception e)
             {
-                throw new ArgumentException("Invalid property");
+                logger.LogDebug(Resources.MSG_IncrementalPRCacheEntryDeserialization, e.Message);
             }
+
+            return cacheEntries;
         }
 
-        private void MultivalueToProps(Dictionary<string, string> props, string settingKey, JArray array)
+        protected static string GetComponentIdentifier(string projectKey, string projectBranch = null)
```

Comment:
This whole thing can be simplified:
* remove Get
* arrow
* ternary
* private
```
        private static string ComponentIdentifier(string projectKey, string projectBranch = null) =>
            string.IsNullOrWhiteSpace(projectBranch) ? projectKey : projectKey + ":" + projectBranch;
```

Reply:
Ha, I though the same when I did the rename.